controllers/datadogagent/feature/logcollection/feature.go

```diff
@@ -0,0 +1,200 @@
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+)
+
+func init() {
+	err := feature.Register(feature.LogCollectionIDType, buildLogCollectionFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildLogCollectionFeature(options *feature.Options) feature.Feature {
+	logCollectionFeat := &logCollectionFeature{}
+
+	return logCollectionFeat
+}
+
+type logCollectionFeature struct {
+	enable                     bool
+	containerCollectAll        bool
+	containerCollectUsingFiles bool
+	containerLogsPath          string
+	podLogsPath                string
+	containerSymlinksPath      string
+	tempStoragePath            string
+	openFilesLimit             int32
+
+	owner metav1.Object
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *logCollectionFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
+	f.owner = dda
+	if dda.Spec.Features.LogCollection != nil && apiutils.BoolValue(dda.Spec.Features.LogCollection.Enabled) {
+		f.enable = true
+
+		if apiutils.BoolValue(dda.Spec.Features.LogCollection.ContainerCollectAll) {
+			f.containerCollectAll = true
+		}
+		if dda.Spec.Features.LogCollection.ContainerCollectUsingFiles == nil || apiutils.BoolValue(dda.Spec.Features.LogCollection.ContainerCollectUsingFiles) {
+			f.containerCollectUsingFiles = true
+		}
+		if dda.Spec.Features.LogCollection.ContainerLogsPath != nil {
+			f.containerLogsPath = *dda.Spec.Features.LogCollection.ContainerLogsPath
+		}
+		if dda.Spec.Features.LogCollection.PodLogsPath != nil {
+			f.podLogsPath = *dda.Spec.Features.LogCollection.PodLogsPath
+		}
+		if dda.Spec.Features.LogCollection.ContainerSymlinksPath != nil {
+			f.containerSymlinksPath = *dda.Spec.Features.LogCollection.ContainerSymlinksPath
+		}
+		if dda.Spec.Features.LogCollection.TempStoragePath != nil {
+			f.tempStoragePath = *dda.Spec.Features.LogCollection.TempStoragePath
+		}
+		if dda.Spec.Features.LogCollection.OpenFilesLimit != nil {
+			f.openFilesLimit = *dda.Spec.Features.LogCollection.OpenFilesLimit
+		} else {
+			f.openFilesLimit = 100
+		}
+	}
+
+	return feature.RequiredComponents{
+		Agent: feature.RequiredComponent{Required: &f.enable},
+	}
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *logCollectionFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) feature.RequiredComponents {
+	f.owner = dda
+
+	if dda.Spec.Features.LogCollection != nil {
+		if apiutils.BoolValue(dda.Spec.Features.LogCollection.Enabled) {
+			f.enable = true
+
+			if apiutils.BoolValue(dda.Spec.Features.LogCollection.LogsConfigContainerCollectAll) {
+				f.containerCollectAll = true
+			}
+			if dda.Spec.Features.LogCollection.ContainerCollectUsingFiles == nil || *dda.Spec.Features.LogCollection.ContainerCollectUsingFiles {
```

Comment:
similarly, this should be defaulted https://github.com/DataDog/datadog-operator/blob/main/apis/datadoghq/v1alpha1/datadogagent_default.go#L754